adb_graphics/figures/maps.py

```diff
@@ -585,5 +595,168 @@ def _xy_mesh(self, field):
+
+    def __init__(self, map_fields, map_, member, model_name=None, **kwargs):
+
+        super().__init__(map_fields, map_, model_name=model_name)
+
+        self.last_panel = kwargs.get('last_panel', False)
+        self.member = str(member)
+
+    def draw(self, show=False):
+
+        ''' Main method for creating the plot. Set show=True to display the
+        figure from the command line. '''
+
+        cf = self._draw_panel(wind_barbs=False)
+
+        self._label_member()
+
+        # Finish with the colorbar on the last panel only
+        # Plot it on the full figure scale.
+        if self.last_panel:
+            cax = plt.axes([0.0, 0.0, 1.0, 0.2])
+            self._colorbar(ax=cax, cc=cf)
+            cax.axis('off')
+
+        # Create a pop-up to display the figure, if show=True
+        if show:
+            plt.tight_layout()
+            plt.show()
+
+        return cf
+
+    def _label_member(self):
+
+        ''' Add the member label to the top left of the plot '''
+
+        ax = self.map.ax
+        ax.text(
+            0.05, 0.90,
+            self.member,
+            fontsize=18,
+            fontweight='bold',
+            backgroundcolor="white",
+            transform=ax.transAxes,
+            )
+
+    def title(self):
+
+        ''' Draw the title for a map. '''
+
+        f = self.field
+        atime = f.date_to_str(f.anl_dt)
+        vtime = f.date_to_str(f.valid_dt)
+        ax = self.map.ax
+
+        # Analysis time (top) and forecast hour with Valid Time (bottom) on the left
+        ax.text(0.0, 0.5,
+                f"{self.model_name}: {atime}\nFcst Hr: {f.fhr}, Valid Time {vtime}",
+                alpha=None,
+                fontsize=14,
+                horizontalalignment='left',
+                verticalalignment='top',
+                transform=ax.transAxes,
+                )
+
+        level, lev_unit = f.numeric_level(index_match=False)
+        if f.vspec.get('print_units', True):
+            units = f'({f.units}, shaded)'
+        else:
+            units = f''
+
+        # Title or Atmospheric level and unit in the high center
+        if f.vspec.get('title'):
+            title = f"{f.vspec.get('title')} {units}"
+        else:
+            level = level if not isinstance(level, list) else level[0]
+            title = f'{level} {lev_unit} {f.field.long_name} {units}'
+        ax.text(0, 0.7,
+                f"{title}",
+                horizontalalignment='left',
+                verticalalignment='top',
+                fontsize=16,
+                transform=ax.transAxes,
+                )
+
+        # Two lines for hatched data (top), and contoured data (bottom) on the right
+        contoured = self._set_overlay_string()
+        ax.text(0, 0.6,
+                f"{contoured}",
+                horizontalalignment='left',
+                verticalalignment='top',
+                fontsize=14,
+                transform=ax.transAxes,
+                )
+
+
+class MapFields():
```

Comment:
This seemed like an okay-ish solution to remove a bunch of duplication in create_graphics for adding contours and hatches. Now we just send the spec, call them all "overlays" and let this more generalized code sort it out. We also can pass it the original "field" object and use it's common info so that we don't have to keep creating other objects with the same parameters.